### PR TITLE
Update SHOW INDEX docs

### LINKED ISSUE #1201: sql: update SHOW INDEX docs to mention Implicit column
```
root@:26257/bank> show index from accounts;
+----------+-------------+--------+-----+---------+-----------+---------+----------+
|  Table   |    Name     | Unique | Seq | Column  | Direction | Storing | Implicit |
+----------+-------------+--------+-----+---------+-----------+---------+----------+
| accounts | primary     | true   |   1 | id      | ASC       | false   | false    |
| accounts | balance_idx | false  |   1 | balance | DESC      | false   | false    |
| accounts | balance_idx | false  |   2 | id      | ASC       | false   | true     |
+----------+-------------+--------+-----+---------+-----------+---------+----------+
(3 rows)
```

### PR DESCRIPTION
**Pull request by jseldess**
- Add Implicit column to all examples
- Other minor fixes

Fixes #1201

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/1271)

### CODE REVIEW

**Comment by jseldess** [comment]
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[drop-index.md, line 49 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhVXz1PVRXzZuyqwxuQ:-KhX3gCR_j8hXBtWqbFd:biqrs5t) ([raw file](https://github.com/cockroachdb/docs/blob/380a8c3c43807f1985a605cb01df8d72f204b0e6/drop-index.md#L49)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

I have a general question about the styling we have here. It looks like we have a separate `~~~ sql` section for a command and its output throughout our docs. This looks fine when there is only a single command and output, but starts to look really cluttered in places like this with repeated commands and outputs alternating. It gets hard to even tell what is input and what is output. Have we considered grouping the command and its output together into a single section?
</blockquote></details>

We actually started separating sql commands and outputs something like a year ago after getting feedback that it's hard to parse when they're in a single block. But we're probably going overboard. 

@sploiselle, what do you think about our updating the guidelines to state that we should provide the sql output only when it's related to the point being made. So no need to show `CREATE TABLE` and such outputs unless necessary?
 https://github.com/cockroachdb/docs/blob/gh-pages/CONTRIBUTING.md#sql-code-samples

---

*[show-index.md, line 34 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhVXqDlr41fCgkRzXm_:-KhX4n-Vo1eP7MTqToMp:b-iu4mim) ([raw file](https://github.com/cockroachdb/docs/blob/380a8c3c43807f1985a605cb01df8d72f204b0e6/show-index.md#L34)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

The following fields are returned for each **column in each** index.
</blockquote></details>

Done. Good catch. Thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1271)*

**Comment by sploiselle** [comment]
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[drop-index.md, line 49 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhVXz1PVRXzZuyqwxuQ:-KhXTbqieHhaqFGOzAzt:bqaj4v4) ([raw file](https://github.com/cockroachdb/docs/blob/380a8c3c43807f1985a605cb01df8d72f204b0e6/drop-index.md#L49)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

We actually started separating sql commands and outputs something like a year ago after getting feedback that it's hard to parse when they're in a single block. But we're probably going overboard. 

@sploiselle, what do you think about our updating the guidelines to state that we should provide the sql output only when it's related to the point being made. So no need to show `CREATE TABLE` and such outputs unless necessary?
 https://github.com/cockroachdb/docs/blob/gh-pages/CONTRIBUTING.md#sql-code-samples
</blockquote></details>

More explicit guidance around that sounds good to me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1271)*

**Comment by jseldess** [comment]
Review status: 0 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[drop-index.md, line 49 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhVXz1PVRXzZuyqwxuQ:-KhYVhZWEkQ2W42Vl4yl:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/380a8c3c43807f1985a605cb01df8d72f204b0e6/drop-index.md#L49)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary><blockquote>

More explicit guidance around that sounds good to me.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1271)*

**Comment by sploiselle** [comment]
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[show-index.md, line 45 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhYkjiI7KQBgvqmicqM:-KhYkjiI7KQBgvqmicqN:b-azs6kr) ([raw file](https://github.com/cockroachdb/docs/blob/f46a8054519820fadeb75c03f0a5bb6914707531/show-index.md#L45)):*
> ```Markdown
> `Direction` | How the column is sorted in the index. Possible values: `ASC` or `DESC` for indexed columns; `N/A` for stored columns.
> `Storing` | Whether or not the `STORING` clause was used to index the column during index creation. Possible values: `true` or `false`.
> `Implicit` | Whether or not the column is an implicit part of the index. Possible values: `true` or `false`.
> ```

I think this could use a little more detail in the description. Not sure which of these questions is best to answer, but I'm left wondering: What does "implicit" mean? Why are we surfacing this information? What's an example of a column that's implicitly indexed? Is there any consequence?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1271)*

**Comment by jseldess** [comment]
*[show-index.md, line 45 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhYkjiI7KQBgvqmicqM:-KhYlenSrGcIZTjuIR-l:bhja929) ([raw file](https://github.com/cockroachdb/docs/blob/f46a8054519820fadeb75c03f0a5bb6914707531/show-index.md#L45)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary><blockquote>

I think this could use a little more detail in the description. Not sure which of these questions is best to answer, but I'm left wondering: What does "implicit" mean? Why are we surfacing this information? What's an example of a column that's implicitly indexed? Is there any consequence?
</blockquote></details>

Good questions. Looking into it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1271)*

**Comment by nvanbenschoten** [comment]
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[drop-index.md, line 49 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhVXz1PVRXzZuyqwxuQ:-KhYt-o76nr6kwXIVcaS:b96sfyt) ([raw file](https://github.com/cockroachdb/docs/blob/380a8c3c43807f1985a605cb01df8d72f204b0e6/drop-index.md#L49)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Done.
</blockquote></details>

That change to the guidelines seems very reasonable to me.

---

*[show-index.md, line 45 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhYkjiI7KQBgvqmicqM:-KhYt8nMlbbvI7NnYrrD:b-2bqvbx) ([raw file](https://github.com/cockroachdb/docs/blob/f46a8054519820fadeb75c03f0a5bb6914707531/show-index.md#L45)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Good questions. Looking into it.
</blockquote></details>

Columns that are implicitly part of an index are those that are part of the index even without having been specified when creating that index. At the moment, I think the only time that this happens is with the primary key column, which is implicitly part of every index that doesn't explicitly include it (in other words, it's part of all indexes whether it's specified or not).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1271)*

**Comment by jseldess** [comment]
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[show-index.md, line 45 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhYkjiI7KQBgvqmicqM:-Khaxkggp5IH5MS6Gv3Z:b-5xw59i) ([raw file](https://github.com/cockroachdb/docs/blob/f46a8054519820fadeb75c03f0a5bb6914707531/show-index.md#L45)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Columns that are implicitly part of an index are those that are part of the index even without having been specified when creating that index. At the moment, I think the only time that this happens is with the primary key column, which is implicitly part of every index that doesn't explicitly include it (in other words, it's part of all indexes whether it's specified or not).
</blockquote></details>

Updated the description. @sploiselle and @nvanbenschoten, let me know what you think. I may add a demonstration (using EXPLAIN), perhaps on the `CREATE INDEX` page.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1271)*

**Comment by sploiselle** [comment]
:lgtm:

---

Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[show-index.md, line 45 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-KhYkjiI7KQBgvqmicqM:-KhbhL7ysYyOiRsnifGu:b-yff50h) ([raw file](https://github.com/cockroachdb/docs/blob/f46a8054519820fadeb75c03f0a5bb6914707531/show-index.md#L45)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Updated the description. @sploiselle and @nvanbenschoten, let me know what you think. I may add a demonstration (using EXPLAIN), perhaps on the `CREATE INDEX` page.
</blockquote></details>

:bulb: awesome

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1271#-:-KhbhPbIEwImmAcIlvtG:bnfp4nl)*
